mod.ts

```diff
@@ -43,15 +39,53 @@ interface Activity {
   }[];
 }
 
+export enum OpCode {
+  HANDSHAKE,
+  FRAME,
+  CLOSE,
+  PING,
+  PONG,
+}
+
 const _ipcHandle = Symbol("[[ipc]]");
 
+function getIPCPath(id: number) {
+  if (id < 0 || id > 9) throw new RangeError(`IPC ID must be between 0-9`);
+
+  let prefix, suffix = `discord-ipc-${id}`;
+  if (Deno.build.os === "windows") {
+    prefix = `\\\\?\\pipe\\`;
+  } else {
+    prefix = Deno.env.get("XDG_RUNTIME_DIR") ?? Deno.env.get("TMPDIR") ??
+      Deno.env.get("TMP") ?? Deno.env.get("TEMP") ?? "/tmp";
+  }
+
+  return `${prefix}${suffix}`;
+}
+
+async function findIPC(id = 0): Promise<string> {
+  const path = getIPCPath(id);
+  try {
+    await Deno.open(path).then((e) => e.close());
+    return path;
+  } catch (e) {
+    return findIPC(id + 1);
+  }
+}
+
 export async function createClient(): Promise<RichPresence> {
+  const path = await findIPC();
   const client = Object.create(RichPresence.prototype);
+
+  // NOTE(DjDeveloperr): Should we remove this condition since OS-specific logic is handled
+  // in the findIPCPath function? This would maybe just start working once unix sockets are
+  // enabled on Windows.
```

Comment:
Let's remove the condition and throw for lack of windows support in `findIPCPath` instead.

Reply:
Done, moved this condition to `findIPCPath`